spec/controllers/admin/ip_based_groups_controller_spec.rb

```diff
@@ -0,0 +1,217 @@
+require 'rails_helper'
+
+RSpec.describe Admin::IpBasedGroupsController, :type => :controller do
+  let(:roles) { [instance_double(Role)] }
+  let(:groups_factory) { class_double(IpBasedGroup) }
+  before do
+    sign_in FactoryGirl.create(:user, :admin)
```

Comment:
I think you're right, but I'm not sure this is the place to fix it.  We are probably best with `sign_in(user) if user`, since that's usable in pretty much any controller spec.  New ticket?